libs/sdk-ui-charts/src/charts/headline/internal/utils/ComparisonDataItemUtils.ts

```diff
@@ -0,0 +1,48 @@
+// (C) 2023 GoodData Corporation
+import { IColorPalette } from "@gooddata/sdk-model";
+import { getRgbStringFromRGB } from "@gooddata/sdk-ui-vis-commons";
+
+import { IColorConfig } from "../../../../interfaces/index.js";
+import { ItemStatus } from "../interfaces/BaseHeadlines.js";
+import {
+    ComparisonColorType,
+    DEFAULT_COMPARISON_PALETTE,
+    getComparisonRgbColor,
+} from "../../headlineHelper.js";
+
+export const getComparisonColor = (
+    colorConfig: IColorConfig,
+    itemStatus: ItemStatus,
+    colorPalette: IColorPalette = DEFAULT_COMPARISON_PALETTE,
+) => {
+    if (colorConfig?.disabled) {
+        return null;
+    }
+
+    const { providedColor, colorType } = getProvidedColorByItemStatus(colorConfig, itemStatus);
+    const rgbColor = getComparisonRgbColor(providedColor, colorType, colorPalette);
+
+    return rgbColor && getRgbStringFromRGB(rgbColor);
+};
+
+const getProvidedColorByItemStatus = (colorConfig: IColorConfig, itemStatus: ItemStatus) => {
+    switch (itemStatus) {
+        case ItemStatus.POSITIVE_VALUE:
+            return {
+                providedColor: colorConfig?.positive,
+                colorType: ComparisonColorType.POSITIVE,
+            };
+
+        case ItemStatus.NEGATIVE_VALUE:
+            return {
+                providedColor: colorConfig?.negative,
+                colorType: ComparisonColorType.NEGATIVE,
+            };
+
+        case ItemStatus.EQUALS_VALUE:
+            return {
+                providedColor: colorConfig?.equals,
+                colorType: ComparisonColorType.EQUALS,
+            };
```

Comment:
There should be a default branch that throws some error about unsupported value.